nixui/state_model.py

```diff
@@ -44,12 +45,20 @@ def get_update_set(self):
     def record_update(self, option, new_value):
         old_value = self.current_values[option]
         if old_value != new_value:
-            self.update_history.append(
-                Update(option, old_value, new_value)
-            )
+            # replace old update if we're still working on the same option
```

Comment:
I think I'd veto against this, at least at its current form. 

1. Undos are only committed on focus lost, which I think is the perfect granularity for this, and
2. if you're testing out different settings for a given value, you could be doing an edit-test loop, and from what I understand, pressing the undo button undos all of the consecutive edits for that option, which is not what I'd expect.

Reply:
When I was smoke testing, I found that the interface was a bit counter-intuitive. `focusOutEvent` didn't always work properly. IIRC, the signal resulting from the undo button being pressed was fired before the `focusOutEvent` was fired. resulting in inconsistent behavior in what "undo" would actually "undo".

There is probably a better way to do this. IMO it should be in a separate issue though, since this works.

Reply:
Oh. Yep, I agree, a separate PR to handle this.